tools/acorn-optimizer.js

```diff
@@ -370,12 +370,22 @@ function runJSDCE(ast, aggressive) {
         ensureData(scopes[scopes.length - 1], node.id.name).def = 1;
       }
       const scope = {};
+      scopes.push(scope);
       node.params.forEach((param) => {
-        const name = param.name;
-        ensureData(scope, name).def = 1;
-        scope[name].param = 1;
+        let param_name;
+        if (param.type === 'RestElement') {
+          param = param.argument;
+        }
+        if (param.type === 'AssignmentPattern') {
+          param_name = param.left.name;
```

Comment:
Not sure if you want to address it here or leave for the solution with escope, but note that left side here can also be any pattern, so it would need to be walked recursively. E.g.:

```js
function foo([a, b] = arguments_) {
}
```

is valid syntax.

Reply:
(if you don't want to address it for now, that's also fine, but maybe worth at least adding an assertion like in the branch below)

Reply:
Done!